pkg/tsdb/influxdb/influxql/influxql.go

```diff
@@ -190,9 +195,25 @@ func execute(ctx context.Context, tracer trace.Tracer, dsInfo *models.Datasource
 	} else {
 		resp = buffered.ResponseParse(res.Body, res.StatusCode, query)
 	}
+
+	if resp.Frames != nil && len(resp.Frames) > 0 {
+		resp.Frames[0].Meta.Custom = readCustomMetadata(res)
+	}
+
 	return *resp, nil
 }
 
+func readCustomMetadata(res *http.Response) map[string]interface{} {
+	result := make(map[string]interface{})
+	for k := range res.Header {
+		if metadataRegex.MatchString(k) {
```

Comment:
Don't think we need the regex. Make `metadataPrefix` lower, and `if key, found := strings.CutPrefix(strings.ToLower(k), metadataPrefix); found { ... }` https://go.dev/pkg/strings/?m=all,old#CutPrefix

Reply:
Please see https://github.com/grafana/grafana/pull/88698/commits/d71da0e2152f0be3566286794bb6c891801d79f3